packages/core/src/metrics/index.js

```diff
@@ -23,7 +23,9 @@ exports.init = _config => {
 /**
  * @typedef {Object} InstanaMetricsModule
  * @property {string} payloadPrefix
- * @property {string} [currentPayload]
+ * // NOTE: the "currentPayload" type is sometimes an array of strings, sometimes an object and sometimes a string.
+ * //       we would need to refactor the currentPayload type to define here a specific type.
```

Comment:
TBD: Do we want to refactor that in 3.0?

Reply:
@kirrg001 I don't think so. Depending on what kind of data we send here, all of the listed options are valid. If it is a simple snapshot attribute (like, the name of the application) a string is the correct type. For the list of dependencies, the type would (correctly) be an object (used as in map-like fashion). Then again, for the process arguments, it would be an array of string.

All of that is fine and understood perfectly fine by the corresponding backend code. Changing any of the types that we send on the wire would introduce a couple of backwards-compatibility issues (newer @instana/collector sending to an older backend or vice versa).

We can instead improve the type though. Either by introducing a simple type alternative here or declaring more explicit types for the different types of metric plug-ins.